recipes/mpfr/all/conanfile.py

```diff
@@ -20,155 +26,174 @@ class MpfrConan(ConanFile):
     options = {
         "shared": [True, False],
         "fPIC": [True, False],
-        "exact_int": ["mpir", "gmp",]
+        "exact_int": ["mpir", "gmp",],
+        "autogen": [True, False],
```

Comment:
Please remove this option and the extra dependencies 🙏

Reply:
Hi @prince-chrismc,

Thanks for your feedback. Certainly, I will do as you request (and we will maintain a private fork of the recipe to meet our needs), but I wanted to clarify a couple of points:
1. These changes work even without using Git source. They simply enable the regeneration of the configure script and Makefiles from their .ac and .in templates. AutotoolsToolchain even has a method for this: Autotools::autoreconf().
2. It is unclear to me how the "git archive" approach would work unless one first runs autoreconf manually before archiving the resulting build area. The benefit of the pending change is that it leverages Conan to automatically ensure one has a self-consistent version of the autotools (autoconf, automake, libtool, etc.) needed to actually be able to run autoreconf to create configure and Makefiles from Git source. In jettisoning this code, CCI is loosing that information and tooling.
3. I think this use case points out an opportunity for a Conan enhancement, which is the ability to add tags to patches in the conandata.yml file. Such a feature would permit selective application of patches on an as needed basis. For example, apply_conandata_patches applies Windows-specific patches regardless of what platform it is running on. Had such a feature existed, I would have used it to selectively apply the autogen patches.
Thanks